### PR TITLE
Add verticalPiano layout

### PR DESCRIPTION
**Pull request by josephktcheung**
This is to address https://github.com/AudioKit/Keyboard/issues/19

* Add `.verticalPiano` layout
* Update demo with VerticalPiano usage

simulator_screenshot_FB987C1D-5671-4EB6-BB16-372AADE45962

### CODE REVIEW

**Comment by josephktcheung** [comment]
Most of logic in https://github.com/AudioKit/Keyboard/blob/bc453c77064fc0d4eb2ba47074dde255549cc5d9/Sources/Keyboard/Layouts/VerticalPiano.swift#L9-L74 is almost identical to those in https://github.com/AudioKit/Keyboard/blob/bc453c77064fc0d4eb2ba47074dde255549cc5d9/Sources/Keyboard/Layouts/Piano.swift#L9-L74. I'm thinking of refactoring them into a struct say `PianoModel` which host those logic.

**Comment by 01GOD** [comment]
Looks nice. Pondering how it could be cool to add a touchbar keyboard to a macOS app. That thin vertical looks about right if rotated 90 degrees. By the way, recommend having a look at https://github.com/AudioKit/PianoRoll also if want to go full on with combining those.

File: Sources/Keyboard/KeyboardKey.swift [hunk 1]
```diff
@@ -36,6 +37,7 @@ public struct KeyboardKey: View {
         self.whiteKeyColor = whiteKeyColor
         self.blackKeyColor = blackKeyColor
         self.pressedColor = pressedColor
+        self.flatLeading = flatLeading
```
Comment (aure) [inline Sources/Keyboard/KeyboardKey.swift]: I'm really not loving the addition of flatLeading to this.

File: Sources/Keyboard/Layouts/Piano.swift [hunk 2]
```diff
@@ -51,14 +50,14 @@ struct Piano<Content>: View where Content: View {
         }
     }
 
-    func whiteKeyWidth(size: CGSize) -> CGFloat {
-        size.width / CGFloat(whiteKeys.count)
+    func whiteKeyDimension(_ dimension: CGFloat) -> CGFloat {
```
Comment (aure) [inline Sources/Keyboard/Layouts/Piano.swift]: I assume this name change is because width implies horizontal, but dimension is just too vague.

Comment (josephktcheung) [reply Sources/Keyboard/Layouts/Piano.swift]: Perhaps we stick with `width`, that means how wide the keyboard key is from side to side.

Screenshot 2022-10-23 at 9 08 19 AM

File: Sources/Keyboard/KeyboardKey.swift (hunk 1 above)
Comment (josephktcheung) [reply Sources/Keyboard/KeyboardKey.swift]: We can keep `flatTop` meaning top of the key is flattened.

File: Sources/Keyboard/Layouts/Piano.swift (hunk 2 above)
Comment (aure) [reply Sources/Keyboard/Layouts/Piano.swift]: Agreed

**Comment by aure** [comment]
For presentation sake, can the vertical piano be on the right side of the demo?

**Comment by josephktcheung** [comment]
> For presentation sake, can the vertical piano be on the right side of the demo?

Updated snapshot in my first comment

File: Sources/Keyboard/KeyboardKey.swift (hunk 1 above)
Comment (josephktcheung) [reply Sources/Keyboard/KeyboardKey.swift]: I use `alignment = .trailing` to determine if the layout is vertical for now, until we have a case when label should be aligned to leading meaning the end of the keys is pointing to the left? Right now the top of the key is always on the left  and end is on the right in vertical layout. Not sure if someone needs it flipped.

Comment (josephktcheung) [reply Sources/Keyboard/KeyboardKey.swift]: Updated padding logic like this:

https://github.com/AudioKit/Keyboard/blob/ee75112dc58fbbe44a1c04fae97ad1b7e15c22cf/Sources/Keyboard/KeyboardKey.swift#L89-L90

Meaning we are flattening the top if we align to bottom, we are flattening the leading if we align to trailing, and vice versa.

**Comment by josephktcheung** [comment]
@aure this can be reviewed again thanks.

**Review by aure — APPROVED** [review]

**Comment by josephktcheung** [comment]
@aure please merge if okay, thanks

**Comment by aure** [comment]
merged and tagged 1.3.5

**Comment by josephktcheung** [comment]
> Looks nice. Pondering how it could be cool to add a touchbar keyboard to a macOS app. That thin vertical looks about right if rotated 90 degrees. By the way, recommend having a look at https://github.com/AudioKit/PianoRoll also if want to go full on with combining those.

@01GOD you can take a look at this comment I made in https://github.com/AudioKit/PianoRoll/pull/5#issuecomment-1288125901, where I show a screen recording of combining the 2 libraries.

**Pull request merged**